proxy/proxy/client.go

```diff
@@ -292,33 +241,129 @@ func isExpired(cfg *tls.Config) bool {
 	return time.Now().After(cfg.Certificates[0].Leaf.NotAfter)
 }
 
-func (c *Client) cachedCfg(instance string) (string, *tls.Config, string) {
+// startRefresh kicks off a refreshCfg asynchronously, returning a channel that is closed
```

Comment:
nit: returning a channel and updating the cacheEntry when it's complete.

Reply:
Not sure what this means. Can you clarify?

Reply:
Just updating the doc string to read a little better.